File: Algorithms/Selection_Algorithms/selection sort.py
def selection_sort(array):       #function to sort using selection sort
    swaps=0
    comparison=0
    for i in range(len(array)):
        mini=-1
        for j in range(i,len(array)):
            if(mini==-1 or array[j]<array[mini]):
                mini=j
            comparison+=1
        swaps+=1
        temp=array[i]
        array[i]=array[mini]
        array[mini]=temp
        print("Array after ",i+1,"th pass is : ",array)
    print("Array after sorting is : ",array)
    print(swaps," number of swaps and",comparison," comparisons")

def input_array():                       #function for taking input of array
    attended=[]
    number_of_attendee=int(input("Enter number of attendee : "))
    for i in range(number_of_attendee):
        tep=int(input("Enter roll number of attendee:"))
        attended.append(tep)
    return attended

if __name__ == '__main__':
    arr=input_array()
    selection_sort(arr)
    print("Array after sorting using selection sort : ",arr)
    

File: Algorithms/Selection_Algorithms/Selection_Sort.go
package main

import (
	"fmt"
	"strconv"
)

func selectionSort(arr [100]int, n int) {
	for i := 0; i < n-1; i++ {
		var minIndex = i
		for j := i + 1; j < n; j++ {
			if arr[minIndex] > arr[j] {
				minIndex = j
			}
		}

		var temp = arr[minIndex]
		arr[minIndex] = arr[i]
		arr[i] = temp
	}
	fmt.Print("After Sorting: ")
	printArray(arr, n)
}

func printArray(arrs [100]int, n int) {
	for i := 0; i < n; i++ {
		fmt.Print(strconv.Itoa(arrs[i]) + " ")
	}
	fmt.Println("")
}

func main() {
	fmt.Print("How many number : ")
	var no = 0
	fmt.Scanln(&no)
	var arr [100]int
	fmt.Print("Enter array values : ")
	for i := 0; i < no; i++ {
		fmt.Scanf("%d", &arr[i])
	}
	fmt.Print("Before Sorting: ")
	printArray(arr, no)
	selectionSort(arr, no)
}
